src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -343,18 +343,31 @@ private async Task SolutionManager_CacheUpdatedAsync(TimeSpan timeSpan, string e
 
         private async ValueTask RefreshWhenNotExecutingActionAsync(RefreshOperationSource source, TimeSpan timeSpanSinceLastRefresh)
         {
-            var sw = Stopwatch.StartNew();
             // Only refresh if there is no executing action. Tell the operation execution to refresh when done otherwise.
             if (_isExecutingAction)
             {
                 _isRefreshRequired = true;
-                EmitRefreshEvent(timeSpanSinceLastRefresh, source, RefreshOperationStatus.NoOp, isUIFiltering: false, duration: 0);
+                EmitRefreshEvent(timeSpanSinceLastRefresh, source, RefreshOperationStatus.NoOp, isUIFiltering: false);
             }
             else
             {
-                await RefreshAsync();
-                sw.Stop();
-                EmitRefreshEvent(timeSpanSinceLastRefresh, source, RefreshOperationStatus.Success, isUIFiltering: false, duration: sw.Elapsed.TotalMilliseconds);
+                var refreshStatus = RefreshOperationStatus.NoOp;
+                var sw = Stopwatch.StartNew();
+                try
```

Comment:
Should we be try catching all refreshes to differentiate between succeeded and failed? Looks like failures were not a consideration when it was originally added.

Reply:
May be a good idea. I just catch here because this was the only place I observed execeptions.